### ISSUE TITLE
Content-Length parsing too strict?

### ISSUE THREAD

**Issue by sebastiansturm** [rust-lang/rust-analyzer#15197]
Hi,

I'm using a fork of Emacs that handles jsonrpc processing on a separate thread, and that version currently emits more than one space character between the 'Content-Length:' header and the actual field value. While I can easily change that fork to emit just a single space, I believe the standard allows for any amount of whitespace (RFC 2616, 4.2?). Also, I think the header names are supposed to be case-insensitive which as far as I can see isn't how the lsp-server crate handles the Content-Length header.

Looks to me like (if I'm interpreting the standard correctly) both would be fixed easily by adding one `trim()` and one `to_lowercase()` within `lsp-server/src/msg.rs` (`read_msg_text`)?
thanks,
Sebastian

**Comment by TomBinford** [comment]
You're correct: https://datatracker.ietf.org/doc/html/rfc9112#name-field-syntax

**Comment by lnicola** [comment]
The LSP spec says:

> Each header field is comprised of a name and a value, separated by ‘: ‘ (a colon and a space).

But right after that we have:

> The structure of header fields conform to the [HTTP semantic](https://tools.ietf.org/html/rfc7230#section-3.2).

Which is pretty ambiguous, notwithstanding the bad grammar.

**Comment by lnicola** [comment]
@sebastiansturm do you want to file a PR with your proposed change?

**Comment by lnicola** [comment]
Actually, it might be better for the whole ecosystem to fix emacs instead.

**Comment by sebastiansturm** [comment]
Emacs is actually fine, it was just my fork that emitted several spaces in a row, and I have fixed that in the meantime so I no longer have any practical issues with rust-analyzer.

I now agree that the lsp spec (I wasn't aware of the corresponding paragraph, thanks for the heads-up) seems to mandate exactly one space. So the only issue that potentially remains is that, according to the spec, the Content-Length header should be case-insensitive? If you agree, I'll gladly prepare a PR for this trivial issue, though if you think things are fine as they are I'm also ok with just closing my bug report.
